modules/external-auth-configuring.adoc

```diff
@@ -0,0 +1,190 @@
+// Module included in the following assemblies:
+//
+// * authentication/external-auth.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="external-auth-configuring_{context}"]
+= Configuring an OIDC identity provider for direct authentication
+
+You can configure {product-title} to directly use an external OIDC identity provider to issue tokens for authentication.
+
+:FeatureName: Direct authentication with an OIDC identity provider
+include::snippets/technology-preview.adoc[]
+
+.Prerequisites
+
+* You have enabled the `TechnologyPreviewNoUpgrade` feature set.
+* You have configured your external authentication provider.
++
+This procedure uses Keycloak as the identity provider and assumes that you have the following clients configured:
+
+** A confidential client for the web console called `console-test` with the valid redirect URIs set to [x-]`https://<openshift_console_route>/auth/callback`
+
+** A public client for the {oc-first} called `oc-cli-test` with the valid redirect URIs set to [x-]`http://localhost:8080`
+
+* You have access to the `kubeconfig` file generated by the installation program for the cluster.
++
+[IMPORTANT]
+====
+Before replacing the built-in OAuth server with an external provider, ensure that you have access to a valid `kubeconfig` file.
```

Comment:
@bergerhoffer , the doc mentioned valid `kubeconfig` file a few times. May this need some clarification? Readers can think an OAuth server token kubeconfig is valid, can't they?

Reply:
@bergerhoffer I don't know how to word it, but a valid kubeconfig file can include cert-based admin kubeconfig, long-lived serviceaccount token kubeconfig file, can't include OAuth server user token or kubeadmin token kubeconfig file.

Reply:
Thanks, let me think on how to adjust this wording